mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb

```diff
@@ -1303,11 +1302,35 @@ func resourceSqlDatabaseInstanceUpdate(d *schema.ResourceData, meta interface{})
 		return err
 	}
 
-	// Update only updates the settings, so they are all we need to set.
-	instance := &sqladmin.DatabaseInstance{
-		Settings: expandSqlDatabaseInstanceSettings(d.Get("settings").([]interface{})),
+	desiredSetting := d.Get("settings")
+	var op *sqladmin.Operation
+	var instance *sqladmin.DatabaseInstance
+	// Major version upgrade
+	if v, ok := d.GetOk("database_version"); ok {
+		instance = &sqladmin.DatabaseInstance{DatabaseVersion: v.(string)}
+		err = retryTimeDuration(func() (rerr error) {
+			op, rerr = config.NewSqlAdminClient(userAgent).Instances.Patch(project, d.Get("name").(string), instance).Do()
```

Comment:
It looks like `resourceSqlDatabaseInstanceUpdate` currently already has a call to `Instances.Update`. I can't tell from the [docs](https://cloud.google.com/sql/docs/mysql/admin-api/rest/v1beta4/instances/update) what the difference between patch and update is (b/241570861) other than that update might restart the instance. It seems like changing the major version would require restarting the instance, so I just wanted to double-check that Patch is sufficient?

Also, would it be possible to standardize resourceSqlDatabaseInstanceUpdate on either using a single Patch call for all updates or using a single Update call for all updates? That would make it easier to maintain.

Reply:
Hi Stephen, when a user updates the `database_version` we actually create a new database instance and copy the resources from the previous version of database, over to the new instance( taking backups in the process ). This operation is atomic and can not be performed with any other update/patch. i.e `database_version` field can not be updated alongside `settings` field. 

Update() does not support `database_version` and Patch() does not support patching any other field in the same request as `database_version`. 

For every other field apart for `database_version` we are using the Update().

I think the code works well in this case, we patch `database_version` then update the rest of the fields.

Reply:
👍 thanks for the additional information.

Reply:
could you add a comment in the code explaining the situation?

Reply:
Added the comment explaining the reason behind the code block.